### PR TITLE
chore: make the latest release from release-vX branch

### PR DESCRIPTION
**Pull request by MartinCupela**
### 🎯 Goal

Make the master branch the pre-release branch and the branch-vX the latest branch.

### CODE REVIEW

File: .releaserc.json [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
+      "channel": "latest"
+    },
+    {
+      "name": "release-v10",
+      "range": "10.x"
+    },
+    {
+      "name": "release-v9",
+      "range": "9.x"
     }
```
Comment (oliverlaz) [inline .releaserc.json:18]: Let's align the `channel` name:
- https://github.com/GetStream/stream-chat-react/blob/release-v10/.releaserc.json
- https://github.com/GetStream/stream-chat-react/blob/release-v9/.releaserc.json

Comment (MartinCupela) [reply .releaserc.json:18]: Thanks for pointing this out

Comment (MartinCupela) [reply .releaserc.json]: I have also updated the channel for "master" to be "rc" to make it consistent with Anton's last pre-releases.

Comment (oliverlaz) [reply .releaserc.json]: nice! thanks for taking care of this!

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -54,7 +54,7 @@ jobs:
     name: Publish documentation from "${{ github.ref_name }}" branch to ${{ inputs.docs_env }}
     runs-on: ubuntu-latest
     # skip during dry runs, release to production only from master, release to staging from anywhere
-    if: ${{ !inputs.dry_run && ((github.ref_name == 'master' && contains('production,staging', inputs.docs_env)) || (github.ref_name != 'master' && inputs.docs_env == 'staging')) }}
+    if: ${{ !inputs.dry_run && ((startsWith(github.ref_name, 'release') && contains('production,staging', inputs.docs_env)) || inputs.docs_env == 'staging') }}
```
Comment (arnautov-anton) [inline .github/workflows/release.yml:57]:
```suggestion
    if: ${{ !inputs.dry_run && ((startsWith(github.ref_name, "release") && inputs.docs_env == "production") || inputs.docs_env == "staging") }}
```

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -24,7 +24,7 @@ jobs:
     name: Release from "${{ github.ref_name }}" branch
     runs-on: ubuntu-latest
     # GH does not allow to limit branches in the workflow_dispatch settings so this here is a safety measure
-    if: ${{ !inputs.docs_only && (github.ref_name == 'master' || github.ref_name == 'rc' || github.ref_name == 'release-v9') }}
+    if: ${{ !inputs.docs_only && startsWith(github.ref_name, 'release') || startsWith(github.ref_name, 'master') }}
```
Comment (arnautov-anton) [inline .github/workflows/release.yml:27]:
```suggestion
    if: ${{ !inputs.docs_only && (startsWith(github.ref_name, 'release') || startsWith(github.ref_name, 'master')) }}
```

**Review by arnautov-anton — APPROVED** [review]

**Pull request merged**